### PR TITLE
Add Energy Based Optimizer

### LINKED ISSUE #4: Implemente agent stage for energy only

### PR DESCRIPTION
**Pull request by rodrigomha**
The model can be built and run but we are still in fixing issues with pipeline of the model in PSI

### CODE REVIEW

**Review by jd-lara — CHANGES_REQUESTED** [review]
Move CSVs to a folder called data or inputs.

File: src/hybrid_build.jl [hunk 1]
```diff
@@ -0,0 +1,297 @@
+import MathOptInterface
```
Comment (jd-lara) [inline src/hybrid_build.jl:1]: Note for future refactor. Implement PSI's `add_variable` methods here. Add comment to that effect

File: src/hybrid_build.jl [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    constraint_status_bid_in =
+        PSI.add_constraints_container!(container, StatusInOn(), HybridSystem, T_rt)
+
+    constraint_status_bid_out =
+        PSI.add_constraints_container!(container, StatusOutOn(), HybridSystem, T_rt)
+
+    constraint_balance =
+        PSI.add_constraints_container!(container, EnergyAssetBalance(), HybridSystem, T_rt)
+
+    # Thermal not implemented yet
```
Comment (jd-lara) [inline src/hybrid_build.jl]: why is this commented out?

File: src/variables_constraints.jl [hunk 3]
```diff
@@ -0,0 +1,36 @@
+using PowerSimulations
```
Comment (jd-lara) [inline src/variables_constraints.jl]: split file for variables and constraints.

**Review by jd-lara — CHANGES_REQUESTED** [review]
Move CSVs to a folder called data or inputs.

File: src/hybrid_build.jl (hunk 1 above)
Comment (rodrigomha) [reply src/hybrid_build.jl:1]: I only need this for adding the objective function. Any way of doing this using PSI?

```
JuMP.@objective(model, MOI.MAX_SENSE, container.objective_function.invariant_terms)
```

Comment (jd-lara) [reply src/hybrid_build.jl:1]: no, you can open an issue in PSI to make a method to this so it can be reused.

**Review by jd-lara — APPROVED** [review]

**Pull request merged**